lib/services/s3.js

```diff
@@ -547,15 +547,15 @@ AWS.util.update(AWS.S3.prototype, {
         regionRedirectErrorCodes.indexOf(error.code) === -1) {
       return done();
     }
-
-    var regionReq = req.service.listObjects({Bucket: bucket, MaxKeys: 0});
+    var reqOperation = AWS.util.isNode() ? 'headBucket' : 'listObjects';
+    var regionReq = req.service[reqOperation]({Bucket: bucket, MaxKeys: 0});
```

Comment:
`MaxKeys` isn't a valid parameter for `headBucket`, so you'll get an `UnexpectedParameter` error here.